src/internal/operators/onErrorResumeNextWith.ts

```diff
@@ -85,41 +81,19 @@ export function onErrorResumeNext<T, A extends readonly unknown[]>(
  * Observable, but - if it errors - subscribes to the next passed Observable
  * and so on, until it completes or runs out of Observables.
  */
-export function onErrorResumeNext<T, A extends readonly unknown[]>(
+export function onErrorResumeNextWith<T, A extends readonly unknown[]>(
   ...sources: [[...ObservableInputTuple<A>]] | [...ObservableInputTuple<A>]
 ): OperatorFunction<T, T | A[number]> {
   // For some reason, TS 4.1 RC gets the inference wrong here and infers the
   // result to be `A[number][]` - completely dropping the ObservableInput part
   // of the type. This makes no sense whatsoever. As a workaround, the type is
   // asserted explicitly.
-  const nextSources = (argsOrArgArray(sources) as unknown) as ObservableInputTuple<A>;
+  const nextSources = argsOrArgArray(sources) as unknown as ObservableInputTuple<A>;
 
-  return operate((source, subscriber) => {
-    const remaining = [source, ...nextSources];
-    const subscribeNext = () => {
-      if (!subscriber.closed) {
-        if (remaining.length > 0) {
-          let nextSource: Observable<A[number]>;
-          try {
-            nextSource = innerFrom<T | A[number]>(remaining.shift()!);
-          } catch (err) {
-            subscribeNext();
-            return;
-          }
-
-          // Here we have to use one of our Subscribers, or it does not wire up
-          // The `closed` property of upstream Subscribers synchronously, that
-          // would result in situation were we could not stop a synchronous firehose
-          // with something like `take(3)`.
-          const innerSub = new OperatorSubscriber(subscriber, undefined, noop, noop);
-          subscriber.add(nextSource.subscribe(innerSub));
-          innerSub.add(subscribeNext);
-        } else {
-          subscriber.complete();
-        }
-      }
-    };
-
-    subscribeNext();
-  });
+  return (source) => oERNCreate(source, ...nextSources);
 }
+
+/**
+ * @deprecated Renamed. Use {@link onErrorResumeNextWith} instead. Will be removed in v8.
+ */
+export const onErrorResumeNext = onErrorResumeNextWith;
```

Comment:
Shouldn't `onErrorResumeNext` stay in this file and `onErrorResumeNextWith` get exported from another file (`onErrorResumeNextWith.ts`)? This is what's done with `combineLatest`, `concat` and others...

Reply:
I'm indifferent about that. That's all internal implementation and has no bearing on users.

Reply:
I think we may have issues with the docs builder, though I'm not completely sure. Let me check it.

Reply:
I can't really make the docs work properly. The code changes from this PR seem to create multiple issues.

One of the issues that this function has is described in #7009.

The other is this:

<img width="1129" alt="image" src="https://user-images.githubusercontent.com/28087049/192164381-ac3f9677-e1a1-41e8-8648-7e34e7cff3ee.png">

The dgeni didn't properly infer the type. The inferred type seems to be: `{ <T, A extends any, unknown>(): any; <T, A extends any, unknown>(): any; <T, A extends any, unknown>(): any; };` which doesn't really mean anything.

I will try to dig deeper into the issue tomorrow.

Reply:
It looks like other constants are having the same issue. E.g. [`flatMap`](https://rxjs.dev/api/index/const/flatMap):

![image](https://user-images.githubusercontent.com/28087049/192457602-467cd778-7a80-4a4a-a913-eb53be259ef8.png)

So, I don't think this is something we can do about by moving to another file, which would be OK only for consistency reasons, nothing else.